src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -817,15 +818,15 @@ private string GenerateFastPathFuncForObject(TypeGenerationSpec typeGenSpec)
                     TypeGenerationSpec propertyTypeSpec = propertyGenSpec.TypeGenerationSpec;
 
                     string runtimePropName = propertyGenSpec.RuntimePropertyName;
+                    string propVarName = propertyGenSpec.PropertyNameVarName;
 
                     // Add the property names to the context-wide cache; we'll generate the source to initialize them at the end of generation.
-                    _currentContext.RuntimePropertyNames.Add(runtimePropName);
+                    _currentContext.RuntimePropertyNames.TryAdd(runtimePropName, propVarName);
```

Comment:
Nit: perhaps include a debug assertion if the given key is pre-populated
```suggestion
                    Debug.Assert(!_currentContext.RuntimePropertyNames.TryGetValue(runtimePropName, out string? existingName) || existingName == propVarName);
                    _currentContext.RuntimePropertyNames.TryAdd(runtimePropName, propVarName);
```

Reply:
Thanks good idea